### PR TITLE
Exit properly on double Ctrl-C when running in watch mode

### PR DESCRIPTION
**Pull request by bengotow**
This PR replaces the bash script I wrote to execute `dagit-cli` or `watchmedo dagit-cli` with a Python script that uses watchdog's API directly.

The problem we observed with the other approach is that the watchmedo CLI attempts to clean up the file watchers BEFORE sending a signal to the subprocess it launches. This process takes a while, and sending a second SIGINT in this time window kills the watchmedo process without a signal ever getting sent to the subprocess.

The relevant code in watchdog is here: https://github.com/gorakhargosh/watchdog/blob/master/src/watchdog/watchmedo.py#L124. This function runs `observer.stop()`, `observer.join()`, and then in the callsite the `handler` is stopped which kills the subprocess.

Thankfully watchdog's python API is pretty nice and watchmedo uses it directly - I basically pulled 20 lines of watchmedo out into our launch script and added the `handler.stop()` call that is missing.

### CODE REVIEW

File: python_modules/dagit/bin/dagit [hunk 1]
```diff
@@ -1,19 +1,41 @@
...
+# Build the dagit-cli command, omitting the --no-watch arg if present
+watch = True
+command = ["dagit-cli"]
+for arg in sys.argv[1:]:
+    if arg == "--no-watch":
+        watch = False
+    else:
+        command.append(arg)
+
+handler = AutoRestartTrick(command=command,
```
Comment (schrockn) [inline python_modules/dagit/bin/dagit]: I think you should leave a comment here that is essentially the description in your PR summary. With a pointer to the original code and why you copy/pasted it here with the modes you made. Also maybe call out the changed lines.

File: python_modules/dagit/bin/dagit [hunk 2]
```diff
@@ -1,19 +1,41 @@
...
+                           patterns=["*.py"],
+                           ignore_patterns=[],
+                           ignore_directories=[],
+                           stop_signal=signal.SIGINT,
+                           kill_after=0.5)
+handler.start()
+
+if watch:
+    print("Watching for file changes...")
+    observer = Observer(timeout=1)
```
Comment (schrockn) [inline python_modules/dagit/bin/dagit]: do timeout=1 and sleep(1) have to be the same number?

**Comment by schrockn** [comment]
Great investigation. 

1) We should leave comments in the file themselves that mirror your PR description here.
2) Should we file an issue in that project? With your code copy/pasted into it?

File: python_modules/dagit/bin/dagit (hunk 1 above)
Comment (bengotow) [reply python_modules/dagit/bin/dagit]: 👍

File: python_modules/dagit/bin/dagit (hunk 2 above)
Comment (bengotow) [reply python_modules/dagit/bin/dagit]: So this is interesting actually - it looks like this timeout is the `Read events blocking timeout / polling interval (in fallback scenarios)` and it defaults to 1, and their source also included a sleep of 1s in the other location. I think that setting this timeout to a value lower than the sleep value may have no effect, but it's hard to tell. Either way these seem to be the defaults they're using in watchmedo so should be OK for now.

**Pull request merged**